src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesProjectTreeProvider.cs

```diff
@@ -0,0 +1,973 @@
+        private OrderPrecedenceImportCollection<IProjectTreePropertiesProvider> ProjectTreePropertiesProviders { get; set; }
+
+        /// <summary>
+        /// Provides a way to extend Dependencies node by consuming sub tree providers that represent
+        /// a particular dependency type and are responsible for Dependencies\[TypeNode] and it's contents.
+        /// </summary>
+        [ImportMany]
+        private OrderPrecedenceImportCollection<IProjectDependenciesSubTreeProvider> SubTreeProviders { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DependenciesProjectTreeProvider"/> class.
+        /// </summary>
+        [ImportingConstructor]
+        public DependenciesProjectTreeProvider(IProjectThreadingService threadingService, 
+                                               UnconfiguredProject unconfiguredProject)
+            : base(threadingService, unconfiguredProject)
+        {
+            ProjectTreePropertiesProviders = new OrderPrecedenceImportCollection<IProjectTreePropertiesProvider>(
+                ImportOrderPrecedenceComparer.PreferenceOrder.PreferredComesLast,
+                projectCapabilityCheckProvider: unconfiguredProject);
+
+            SubTreeProviders = new OrderPrecedenceImportCollection<IProjectDependenciesSubTreeProvider>(
+                                    ImportOrderPrecedenceComparer.PreferenceOrder.PreferredComesLast,
+                                    projectCapabilityCheckProvider: unconfiguredProject);
+        }
+
+        /// <summary>
+        /// See <see cref="IProjectTreeProvider"/>
+        /// </summary>
+        /// <remarks>
+        /// This stub defined for code contracts.
+        /// </remarks>
+        IReceivableSourceBlock<IProjectVersionedValue<IProjectTreeSnapshot>> IProjectTreeProvider.Tree
+        {
+            get { return Tree; }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether a given set of nodes can be copied or moved underneath some given node.
+        /// </summary>
+        /// <param name="nodes">The set of nodes the user wants to copy or move.</param>
+        /// <param name="receiver">
+        /// The target node where <paramref name="nodes"/> should be copied or moved to.
+        /// May be <c>null</c> to determine whether a given set of nodes could allowably be copied anywhere (not 
+        /// necessarily everywhere).
+        /// </param>
+        /// <param name="deleteOriginal"><c>true</c> for a move operation; <c>false</c> for a copy operation.</param>
+        /// <returns><c>true</c> if such a move/copy operation would be allowable; <c>false</c> otherwise.</returns>
+        public override bool CanCopy(IImmutableSet<IProjectTree> nodes, 
+                                     IProjectTree receiver, 
+                                     bool deleteOriginal = false)
+        {
+            return true;
```

Comment:
this actually should be always false , i changed it. 

---
In reply to: [74323746](https://github.com/dotnet/roslyn-project-system/pull/349#discussion_r74323746) [](ancestors = 74323746)